### PR TITLE
Fix #1780  -  let dynamic intellisense have per-invoke http channel

### LINKED ISSUE #1780: Intellisense: dynamic intellisense needs httpClient per request
Dynamic intellisense is using a global http client provided by Config.AddService. But these may be short lived clients. 
Instead, it should take it per request. Each EditorContextScope should have its own http client. 

This is used for the call to ServiceFunction.GetConnectorSuggestionsAsync

Also see #1759

### PR DESCRIPTION
**Pull request by MikeStall**
Fix #1780 

Intellisense can invoke Http calls for dynamic intellisense. 

Http Channels can be short lived.

Allow the EditorContextScope to pass in a new context object for connectors to help control them. 
This has to be plumbed through the whole stack:

1. EditorContextScope
2. Engine.Suggest
3. IntellisenseContext / IntellisenseData
4. TexlFunction.GetConnectorSuggestionsAsync
5. ServiceFunction.
6. ScopedHttpFunctionInvoker / HttpFunctionInvoker
7. make the actual http call.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Connectors.Tests/IntellisenseTests.cs [hunk 1]
```diff
@@ -167,6 +168,89 @@ public void ConnectorIntellisenseTest2(int responseIndex, int networkCall, strin
...
+
+            var ctx = new TestRuntimeConnectorContext();
+            ctx._clients[cxNamespace] = client;
+            BasicServiceProvider services = new BasicServiceProvider();
+            services.AddService<RuntimeConnectorContext>(ctx);
+                        
+            IPowerFxScope scope = new EditorContextScope(
+                (expr) => engine.Check(expression, symbolTable: null))
+            {
+                Services = services
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/IntellisenseTests.cs:220]: Rance / Massimo - does this test case solve your issue?

File: src/tests/Microsoft.PowerFx.Connectors.Tests/IntellisenseTests.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@
 using Microsoft.OpenApi.Models;
 using Microsoft.PowerFx.Intellisense;
 using Microsoft.PowerFx.Tests;
+using Newtonsoft.Json.Serialization;
```
Comment (jas-valgotar) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/IntellisenseTests.cs:12]: Newtonsoft?

File: src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs [hunk 3]
```diff
@@ -241,7 +241,7 @@ protected void ValidateArgumentIsMutable(TexlBinding binding, TexlNode arg, IErr
 
         // This method is used for managing "x-ms-dynamic-values" and "x-ms-dynamic-schema" OpenApi extensions in connectors
         // https://learn.microsoft.com/en-us/connectors/custom-connectors/openapi-extensions
-        public virtual async Task<ConnectorSuggestions> GetConnectorSuggestionsAsync(FormulaValue[] knownParameters, int argPosition, CancellationToken cts)
+        public virtual async Task<ConnectorSuggestions> GetConnectorSuggestionsAsync(FormulaValue[] knownParameters, int argPosition, IServiceProvider services, CancellationToken cts)
```
Comment (jas-valgotar) [inline src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs:244]: > IServiceProvider services, [](http://example.com/codeflow?start=133&length=27)

Can this break PA?

**Review by jas-valgotar — APPROVED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Connectors.Tests/IntellisenseTests.cs (hunk 2 above)
Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Connectors.Tests/IntellisenseTests.cs:12]: D'oh - VS auto added

File: src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs (hunk 3 above)
Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs:244]: we should be ok - PA doesn't implement this virtual.

File: src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs [hunk 4]
```diff
@@ -227,14 +228,15 @@ private async Task<FormulaValue> ExecuteHttpRequest(string cacheScope, bool thro
 
             return await _cache.TryGetAsync(cacheScope, key, async () =>
             {
-                var response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                var client = localInvoker ?? _httpClient;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs:231]: >   var client = localInvoker ?? _httpClient; [](http://example.com/codeflow?start=13&length=44)

This is the key line to merge conflgs

**Review by jas-valgotar — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**